src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -177,7 +178,16 @@ public function get(): ?array
 
             // Append pessimistic write lock to FROM clause if db platform supports it
             $sql = $query->getSQL();
-            if (preg_match('/FROM (.+) WHERE/', (string) $sql, $matches)) {
+
+            if (method_exists(QueryBuilder::class, 'forUpdate')) {
+                $query->forUpdate();
+                try {
+                    $sql = $query->getSQL();
+                } catch (Exception $e) {
```

Comment:
can this be a bit more specific? I don't like swallowing any exceptions :grimacing:

Reply:
!
```suggestion
                } catch (DBALException $e) {
```

Reply:
https://github.com/doctrine/dbal/pull/6191#discussion_r1379382594